csharp/lang/security/filesystem/unsafe-path-combine.yaml

```diff
@@ -0,0 +1,40 @@
+rules:
+  - id: unsafe-path-combine
+    mode: taint
+    pattern-sources:
+      - patterns:
+          - pattern: $ARG
+          - pattern-inside: |
+                public $T $M(...,string $ARG,...){...}
+    pattern-sinks:
+      - patterns: 
+        - pattern: $ARG
+        - pattern-inside: |
+            Path.Combine(...)
+    pattern-sanitizers:
+      - patterns:
+        - pattern: $ARG
+        - pattern-either:
+          - pattern-inside: |
+              $ARG = Path.GetFileName(...);
+              ...
+          - pattern-inside: |
+              <... Path.GetFileName(...) != $ARG ...>;
```

Comment:
PA team has pointed out that this approach is quite slow - maybe we just accept that we don't catch this case?

Reply:
Could we measure how slow? Is it like, screeching halt kind of slow? Or step away to make coffee kind of slow?

Reply:
it's "get a coffee" slow, but this is uselessly noisy on a real project:

```
PS /Users/kboberg/source/Squirrel.Windows> measure-command {semgrep --config s/returntocorp:unsafe-path-combine}
Fetching rules from https://semgrep.dev/registry ...
Running 1 rules...
ran 1 rules on 43 files: 21 findings

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 2
Milliseconds      : 441
Ticks             : 24417182
TotalDays         : 2.82606273148148E-05
TotalHours        : 0.000678255055555556
TotalMinutes      : 0.0406953033333333
TotalSeconds      : 2.4417182
TotalMilliseconds : 2441.7182
```

Reply:
That's not too bad 👍

Reply:
There was an edge case I wasn't handling, fixed and the results are better now.  Turns out a CLI client app is uh, kinda lousy with path traversal because 🤷